### ISSUE TITLE
Collections: Dashboard -> Collections list should not instantiate a collection for each one in the list

### ISSUE THREAD

**Issue by elrayle** [samvera/hyrax#1903]
### Descriptive summary

Make collections abilities have the option of only passing in the id or solr doc to avoid having to get the collection object from Fedora.

Targeted usage:  Dashboard -> Collections -> list of collections

This page currently creates an instance of collection to be able to check if the user can deposit in the collection.  Once permissions service is refactored to allow collection id instead of collection instance AND collection ability is updated to check deposit ability using the id, this page will no longer need the collection instance.

### Rationale

Avoid having to create an instance of collection when later the only thing we use is the id. The id can come from the solr doc which is more efficient.

### Implementation

Update `app/models/concerns/hyrax/ability/collection_ability.rb`

- [x] update the first deposit ability for Collection
- [x] add the second deposit ability for String

```
        can :deposit, Collection do |collection|
          Hyrax::Collections::PermissionsService.can_deposit_in_collection?(user: current_user, collection_id: collection.id)
        end
        can :deposit, String do |id|
          Hyrax::Collections::PermissionsService.can_deposit_in_collection?(user: current_user, collection_id: id)
        end
```

Update `app/views/hyrax/my/_collection_action_menu.html.erb`
- [x] delete line that creates the instance of collection `<% collection = Collection.find collection_presenter.id %>`
- [x] update line check deposit access

```
ORIGINAL:     hasaccess: (can? :deposit, collection_presenter) } do %>
REFACTOR:     hasaccess: (can? :deposit, collection_presenter.id) } do %>
```

### Steps to reproduce the behavior

1. Create a collection
1. Dashboard -> Collections

Verify that after the abilities change, the list still checks permissions correctly when displaying the action menu.

### Dependency

#1902 Collections: refactor permissions services to take id as parameter whenever possible

**Comment by elrayle** [comment]
It was not reasonable to have abilities check directly on the id.  To do that requires...
```
can? :deposit, String do |id|
```
String matches too many times and too generally.

Instead, changed to check solr doc instead.  We can get the id from the solr doc.
```
can? :deposit, SolrDocument do |solr_doc|
```

**Comment by elrayle** [comment]
Partial fix by PR #1904 to update abilities.
Partial fix by PR #1812 for updates to collection action menu.

The two PRs meet all the requirements of this issue.
